app/api/resources/user.py

```diff
@@ -508,3 +508,99 @@ def get(cls):
             return messages.USER_NOT_FOUND, HTTPStatus.NOT_FOUND
 
         return dashboard, HTTPStatus.OK
+
+@users_ns.route("user/forgot_password")
+class ForgotPassword(Resource):
+    @classmethod
+    @users_ns.doc(f"forgot password")
+    @users_ns.response(HTTPStatus.OK, f"{messages.PASSWORD_RESET_MAIL_MESSAGE}")
+    @users_ns.response(HTTPStatus.BAD_REQUEST, f"{messages.EMAIL_FIELD_IS_MISSING}"'\n'f"{messages.EMAIL_INPUT_BY_USER_IS_INVALID}")
+    @users_ns.response(HTTPStatus.FORBIDDEN, f"{messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN}")
+    @users_ns.response(HTTPStatus.NOT_FOUND, f"{messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM}")
+    @users_ns.expect(forgot_password_change_request_data_model, validate=True)
```

Comment:
doesn't this do the same as your line 534 and 535? If you remove those lines, won't you get an error anyways for the missing field?

Reply:
I have tried removing those lines 534 & 535 lines, and it will surely show that **email is invalid** message but without those lines, it won't show you the exact **missing field** message [which will be helpful for the user to know what exactly to error what to say ], I hope this clears the sky 

![2020-07-04](https://user-images.githubusercontent.com/36916536/86509862-bd1c8480-be08-11ea-9fe1-592bfe105920.png)

Reply:
and @isabelcosta  if I won't remove those lines I would get the following descriptive & helpful message:-

![2020-07-04 (1)](https://user-images.githubusercontent.com/36916536/86509915-3c11bd00-be09-11ea-9143-c1116922ccf1.png)